crates/ty_python_semantic/src/types.rs

```diff
@@ -2709,6 +2709,22 @@ impl<'db> Type<'db> {
                     )
                 })
                 .unwrap_or_else(|| ConstraintSet::from(relation.is_assignability())),
+
+            // Similarly, `Literal[<class 'C'>]` is assignable to `C[...]` (a generic-alias type)
```

Comment:
`C[...]` is the display we use for "instance of a specialised version of `C`", not "the specialised version of `C` itself"

```suggestion
            // Similarly, `<class 'C'>` is assignable to `<class 'C[...]'>` (a generic-alias type)
```